CAM-11925 added support for file variables with spaces in their name

johnccr opened the pull request:
Please review the code changes I made to fix the issue related to truncated filenames when downloading file variables which filenames contain spaces. filename attribute of content-disposition should enclose the file name in quotes in order for browsers to properly get the name.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/camunda/camunda-bpm-platform?pullRequest=834) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/camunda/camunda-bpm-platform?pullRequest=834) before we can accept your contribution.<br/><hr/>**Juan C Calderon** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/camunda/camunda-bpm-platform?pullRequest=834) it.</sub>

tasso94 commented:
Hi @johnccr,

Sorry for my late response. 

I understand the problem, and I can confirm that this is unexpected behavior. I've updated the ticket description of CAM-11925 with my findings. However, before I can merge your code changes, we need to make sure that the automated tests are adjusted accordingly. 

The following test classes need to be adjusted:
* `CaseDefinitionRestServiceInteractionTest`
* `DecisionDefinitionRestServiceInteractionTest`
* `DecisionRequirementsDefinitionRestServiceInteractionTest`
* `DeploymentRestServiceInteractionTest`
* `ProcessDefinitionRestServiceInteractionTest`
* `VariableInstanceRestServiceInteractionTest`
* `HistoricDetailRestServiceInteractionTest`
* `HistoricVariableInstanceRestServiceInteractionTest`

Please let me know if you have any questions.

Best,
Tassilo

johnccr commented:
Hello Tassilo,
I will make the changes on those and let you know when done.

Regards,
Juan Carlos Calderón Rojas
 

    On Tuesday, May 26, 2020, 9:26:16 AM CDT, Tassilo Weidner <notifications@github.com> wrote:  
 
 


Hi @johnccr,

Sorry for my late response.

I understand the problem, and I can confirm that this is unexpected behavior. I've updated the ticket description of CAM-11925 with my findings. However, before I can merge your code changes, we need to make sure that the automated tests are adjusted accordingly.

The following test classes need to be adjusted:
   
   - CaseDefinitionRestServiceInteractionTest
   - DecisionDefinitionRestServiceInteractionTest
   - DecisionRequirementsDefinitionRestServiceInteractionTest
   - DeploymentRestServiceInteractionTest
   - ProcessDefinitionRestServiceInteractionTest
   - VariableInstanceRestServiceInteractionTest
   - HistoricDetailRestServiceInteractionTest
   - HistoricVariableInstanceRestServiceInteractionTest

Please let me know if you have any questions.

Best,
Tassilo

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

johnccr commented:
Hello Tassilo,
I have modified all these files and a few others that also download files. I ammendded the commit on Borealix repo, not sure if the pull requests register those changes. Let me know if there is anything else on my side.

Regards,
Juan Carlos Calderón Rojas
 

    On Tuesday, May 26, 2020, 9:26:16 AM CDT, Tassilo Weidner <notifications@github.com> wrote:  
 
 


Hi @johnccr,

Sorry for my late response.

I understand the problem, and I can confirm that this is unexpected behavior. I've updated the ticket description of CAM-11925 with my findings. However, before I can merge your code changes, we need to make sure that the automated tests are adjusted accordingly.

The following test classes need to be adjusted:
   
   - CaseDefinitionRestServiceInteractionTest
   - DecisionDefinitionRestServiceInteractionTest
   - DecisionRequirementsDefinitionRestServiceInteractionTest
   - DeploymentRestServiceInteractionTest
   - ProcessDefinitionRestServiceInteractionTest
   - VariableInstanceRestServiceInteractionTest
   - HistoricDetailRestServiceInteractionTest
   - HistoricVariableInstanceRestServiceInteractionTest

Please let me know if you have any questions.

Best,
Tassilo

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

tasso94 commented:
Hi @johnccr,

The pull request registered your latest changes. I'll have a look and give feedback or merged it soon.

Best,
Tassilo

tasso94 commented:
Hi @johnccr,

I just run the test suite in `engine-rest/engine-rest` with `mvn clean install` and there are still failing tests:
* `CaseDefinitionRestServiceInteractionTest#testCaseDiagramNullFilename`
* ` DecisionDefinitionRestServiceInteractionTest#testDecisionDiagramNullFilename`
* `ProcessDefinitionRestServiceInteractionTest#testProcessDiagramNullFilename`

Could you please fix these tests as well?

Best,
Tassilo

johnccr commented:
Hey Tassilo,
Let me check, because I was not sure what to do with those "null file" test cases. And I cannot do a "mvn install" due to some test cases failing to compile on DMN project :/. I'll try to replicate the test case execution on rest-engine only and if I succeed I will send a new update today by the end of the day.

Regards,
Juan C Calderón
   On Wednesday, May 27, 2020, 4:09:51 AM CDT, Tassilo Weidner <notifications@github.com> wrote:  
 
 


Hi @johnccr,

I just run the test suite in engine-rest/engine-rest with mvn clean install and there are still failing tests:
   
   - CaseDefinitionRestServiceInteractionTest#testCaseDiagramNullFilename
   -  DecisionDefinitionRestServiceInteractionTest#testDecisionDiagramNullFilename
   - ProcessDefinitionRestServiceInteractionTest#testProcessDiagramNullFilename

Could you please fix these tests as well?

Best,
Tassilo

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

johnccr commented:
Hello again Tassilo
I was able to run the test cases for rest-engine, and I was able to fix the test cases you mentioned below. Changes were pushed to github, please check again when you can.

Regards,
Juan Carlos Calderón Rojas
 

    On Wednesday, May 27, 2020, 4:09:51 AM CDT, Tassilo Weidner <notifications@github.com> wrote:  
 
 


Hi @johnccr,

I just run the test suite in engine-rest/engine-rest with mvn clean install and there are still failing tests:
   
   - CaseDefinitionRestServiceInteractionTest#testCaseDiagramNullFilename
   -  DecisionDefinitionRestServiceInteractionTest#testDecisionDiagramNullFilename
   - ProcessDefinitionRestServiceInteractionTest#testProcessDiagramNullFilename

Could you please fix these tests as well?

Best,
Tassilo

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

tasso94 commented:
Hi @johnccr,

Thank you for your contribution again!

I've merged your changes (2a02cf1) with our master branch. 🎉 

Best,
Tassilo

johnccr commented:
Great Tassilo, happy to help.
